torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -692,6 +702,16 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // Mutex to guard maps like devNCCLCommMap_ and ncclIdToCommMap_.
   std::mutex mutex_;
 
+  // Heartbeat of watchdog thread.
+  std::atomic<int> heartbeat_;
```

Comment:
i think we do not want an atomic here.  I am not sure if it is well defined whether an atomic will use a lock or not.  (at best, it may depend on the processor you are compiling to as well as the type (<int>).

We don't really need a lock here.  One thread is always writing, the other thread is always reading.  You may want to look into volatile.

Reply:
I thought about it, too. Since only one rewriter and one reader, using atomic is an overkill.